### ISSUE TITLE
add arg to "wallet spend" to not aggressively spend *all* spendable outputs

### ISSUE THREAD

**Issue by antiochp** [mimblewimble/grin#208]
Currently the coin selection logic grabs *all* outputs that are spendable based on the number of confirmations (defaults to 1 confirmation).
This means we actually prevent the wallet being able to generate more than one tx for a given block (unless we allow explicitly allow the wallet to spend 0-conf outputs).

It might be useful to add an additional flag/arg to `wallet spend` to limit how many outputs it selects or change the selection criteria somehow.
Artificially restricting the wallet to "1 tx per block" by default may end up causing more confusion than it is worth without there being an easy flag to pass to change this behavior.

__Pros__ for grabbing all spendable txs: 
* keeps the utxo set small

__Cons__ for grabbing all spendable txs:
* locks them all up for the time it takes to confirm the tx

**Comment by antiochp** [comment]
Resolved in #256 

`wallet send` (defaults to spending _all_ eligible coins).
`wallet send --selection smallest` (will spend minimum number of small coins first, leaving larger coins unspent if possible).

**Comment by antiochp** [comment]
```
USAGE:
    grin wallet send [OPTIONS] [amount]

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

OPTIONS:
    -d, --dest <dest>                         Send the transaction to the provided server
    -c, --min_conf <minimum_confirmations>    Minimum number of confirmations required for an output to be spendable.
                                              [default: 1]
    -s, --selection <selection_strategy>      Coin/Output selection strategy (default|smallest) [default: default]
                                              [values: default, smallest]

ARGS:
    <amount>    Number of coins to send with optional fraction, e.g. 12.423
```
